src/amltk/sklearn/evaluation.py

```diff
@@ -793,6 +886,102 @@ class CVEvaluation(EvaluationProtocol):
     data.
     """
 
+    class FoldInfo(NamedTuple):
+        """Information about a fold.
+
+        Args:
+            trial: The trial that is being evaluated.
+            pipeline: The pipeline being evaluated.
+            fold: The fold number that was just evaluated
```

Comment:
Also a larger point: do we want to rename/reframe this as `SplitInfo` instead of `FoldInfo`?

If I want to perform repeated cross-validation or repeated shuffle splits, I would still like to do early stopping. 

I think so far we only talked about CVEarlySopping but in my mind, this should be abstracted to EvaluationEarlyStopping in a more general form. All assume the same base format and structure (a list of scores) just the names are a bit different.

Reply:
I changed it to split info. I have no strong feelings one way or the other and `Split` is a bit more general. I will still call it all based of `CVxxx` due to the evaluation method being called `CVEValuation`. Not sure how to make it more general or even if we want to do so. The less assumptions made, the less that can be automated or the more configuration the user is required to do.

Reply:
Maybe repeated evaluation or something like that, but that is something for the future.